lib/curl_ntlm_core.c

```diff
@@ -78,7 +78,16 @@
 #  include <wolfssl/openssl/rand.h>
 #endif
 
-#  if (defined(OPENSSL_VERSION_NUMBER) && \
+#  if defined(USE_WOLFSSL) && defined(OPENSSL_COEXIST)
+#    define DES_key_schedule WOLFSSL_DES_key_schedule
+#    define DES_cblock WOLFSSL_DES_cblock
+#    define DES_set_odd_parity wolfSSL_DES_set_odd_parity
+#    define DES_set_key wolfSSL_DES_set_key
+#    define DES_set_key_unchecked wolfSSL_DES_set_key_unchecked
+#    define DES_ecb_encrypt wolfSSL_DES_ecb_encrypt
+#    define DESKEY(x) ((WOLFSSL_DES_key_schedule *)(x))
+#    define DESKEYARG(x) *x
```

Comment:
To work in MultiSSL mode (pending #15596), the order of
`#if` branches is significant. wolfSSL should follow OpenSSL,
according to the comment at the top of the file.

I haven't double-checked this with a build, but it might
be something to verify or adjust.

Reply:
These patches resolve the above and a checksrc issue:
https://github.com/curl/curl/pull/15596/commits/912bd7f8c3f37caee3b87be63d98dbd1f069069d
https://github.com/curl/curl/pull/15596/commits/aef2556dfcb510a72a53925661feba75dd47963d

Tested via #15596. The MultiSSL case isn't CI tested yet, that one I did locally.